tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/MsTestV2/TestMethodRunnerExecuteIntegration.cs

```diff
@@ -44,24 +46,70 @@ public static class TestMethodRunnerExecuteIntegration
             return new CallTargetReturn<TReturn>(returnValue);
         }
 
-        if (returnValue is Array { Length: 1 } returnValueArray)
+        if (returnValue is IList { Count: > 0 } lstResults)
         {
-            var unitTestResultObject = returnValueArray.GetValue(0);
-
-            if (unitTestResultObject != null && unitTestResultObject.TryDuckCast<UnitTestResultStruct>(out var unitTestResult))
+            foreach (var unitTestResultObject in lstResults)
             {
-                if (unitTestResult.Outcome is UnitTestResultOutcome.Inconclusive or UnitTestResultOutcome.NotRunnable or UnitTestResultOutcome.Ignored)
+                if (unitTestResultObject.TryDuckCast<UnitTestResultStruct>(out var unitTestResult))
                 {
-                    if (!MsTestIntegration.ShouldSkip(instance.TestMethodInfo, out _, out _))
+                    if (unitTestResult.Outcome is UnitTestResultOutcome.Inconclusive or UnitTestResultOutcome.NotRunnable or UnitTestResultOutcome.Ignored)
+                    {
+                        if (!MsTestIntegration.ShouldSkip(instance.TestMethodInfo, out _, out _))
+                        {
+                            // This instrumentation catches all tests being ignored
+                            MsTestIntegration.OnMethodBegin(instance.TestMethodInfo, instance.GetType())?
+                               .Close(TestStatus.Skip, TimeSpan.Zero, unitTestResult.ErrorMessage);
+                        }
+                    }
+                    else if (unitTestResult.Outcome is UnitTestResultOutcome.Error or UnitTestResultOutcome.Failed)
                     {
-                        // This instrumentation catches all tests being ignored
-                        MsTestIntegration.OnMethodBegin(instance.TestMethodInfo, instance.GetType())?
-                                         .Close(TestStatus.Skip, TimeSpan.Zero, unitTestResult.ErrorMessage);
+                        // We need to check if the test is failing because a Class initialization error
+                        if (instance.TestMethodInfo.Parent.Instance is { } classInfoObject &&
```

Comment:
technically can omit the explicit null check as we check inside TryDuckCast<T> anyway

Reply:
true, changed in [f4d4b7a](https://github.com/DataDog/dd-trace-dotnet/pull/5381/commits/f4d4b7acfdfab96cfbb9680c8ef1299db40978f7)